scipy/integrate/_quadrature.py

```diff
@@ -627,26 +654,115 @@ def simpson(y, x=None, dx=1.0, axis=-1, even='avg'):
         if x.shape[axis] != N:
             raise ValueError("If given, length of x along axis must be the "
                              "same as y.")
+
+    # even keyword parameter is deprecated
+    if even is not None:
+        warnings.warn(
+            "The 'even' keyword is deprecated as of SciPy 1.11.0 and will be "
+            "removed in SciPy 1.13.0",
+            DeprecationWarning, stacklevel=2
+        )
+
     if N % 2 == 0:
         val = 0.0
         result = 0.0
-        slice1 = (slice(None),)*nd
-        slice2 = (slice(None),)*nd
-        if even not in ['avg', 'last', 'first']:
-            raise ValueError("Parameter 'even' must be "
-                             "'avg', 'last', or 'first'.")
+        slice_all = (slice(None),) * nd
+
+        # default is 'simpson'
+        even = even if even is not None else "simpson"
+
+        if even not in ['avg', 'last', 'first', 'simpson']:
+            raise ValueError(
+                "Parameter 'even' must be 'simpson', "
+                "'avg', 'last', or 'first'."
+            )
+
+        if N == 2:
```

Comment:
Actually looking again should this be inside the `if even == 'simpson':` otherwise `val += 0.5 * last_dx * (y[slice1] + y[slice2])`  is applied twice for avg, first and last?

Reply:
Oh wait sorry missed the line setting `even = None`

Reply:
It was structured like this because `even='avg'` needs three things. The 
If we use the approach you suggested we'd have to increase the indentation level of the code in the `else` block by 1.

Reply:
Ok fine let's leave it as is